src/openforms/forms/api/viewsets.py

```diff
@@ -254,7 +254,7 @@ def get_queryset(self):
         queryset = super().get_queryset()
         if not self.request.user.is_staff:
             # if not staff then only show active
-            queryset = queryset.filter(active=True)
+            queryset = queryset.filter(_is_deleted=False).filter(active=True)
```

Comment:
I think we should only return deleted forms for the detail endpoint (`self.action == "retrieve"`) and not list endpoints (`self.action == "list"`).

Third parties (authenticated) can integrate displaying a dropdown of forms to embed in their CMS - you only want to see non-deleted forms there. And the admin uses the detail/retrieve endpoint and does not use the list endpoint.